pkg/build/sbom.go

```diff
@@ -0,0 +1,51 @@
+// Copyright Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build
+
+import (
+	"fmt"
+	"path"
+
+	"istio.io/pkg/log"
+	"istio.io/release-builder/pkg/model"
+	"istio.io/release-builder/pkg/util"
+)
+
+// Sbom generates Software Bill Of Materials for istio repo in an SPDX readable format.
+func GenerateBillOfMaterials(manifest model.Manifest) error {
+	// Retrieve istio repository path to run the sbom generator
+	istioRepoDir := manifest.RepoDir("istio")
+	sourceSbomFile := path.Join(manifest.OutDir(), "istio-source.spdx")
+	sourceSbomNamespace := fmt.Sprintf("https://storage.googleapis.com/istio-release/releases/%s/istio-source.spdx",
+		manifest.Version)
+	releaseSbomFile := path.Join(manifest.OutDir(), "istio-release.spdx")
+	releaseSbomNamespace := fmt.Sprintf("https://storage.googleapis.com/istio-release/releases/%s/istio-release.spdx",
```

Comment:
OK, we don't (yet) have a way to link these two from the command line (you can do it from the library). Hopefully, we can tackle that in an upcoming release of `bom`.